benchmarks/distributed/rpc/rl/launcher.py

```diff
@@ -0,0 +1,113 @@
+import argparse
+import os
+import pdb
+
+import torch
+import torch.distributed.rpc as rpc
+import torch.multiprocessing as mp
+import json
+import numpy as np
+
+import time
+
+from coordinator import CoordinatorBase
+
+COORDINATOR_NAME = "coordinator"
+AGENT_NAME = "agent"
+OBSERVER_NAME = "observer{}"
+
+TOTAL_EPISODES = 10
+TOTAL_EPISODE_STEPS = 100
+
+
+def str2bool(v):
+    if isinstance(v, bool):
+        return v
+    if v.lower() in ('yes', 'true', 't', 'y', '1'):
+        return True
+    elif v.lower() in ('no', 'false', 'f', 'n', '0'):
+        return False
+    else:
+        raise argparse.ArgumentTypeError('Boolean value expected.')
+
+
+parser = argparse.ArgumentParser(description='PyTorch RPC RL Benchmark')
+parser.add_argument('--world_size', type=int, default=5)
+parser.add_argument('--master_addr', type=str, default='127.0.0.1')
+parser.add_argument('--master_port', type=str, default='29501')
+parser.add_argument('--batch', type=str2bool, default=True)
+
+parser.add_argument('--state_size', type=str, default='10,20,10')
+parser.add_argument('--nlayers', type=int, default=5)
+parser.add_argument('--out_features', type=int, default=10)
+parser.add_argument('--graph_variable', type=str, default='')
+
+args = parser.parse_args()
+args = vars(args)
+
+def run_worker(rank, world_size, master_addr, master_port, batch, state_size, nlayers, out_features, queue=None):
+    state_size = list(map(int, state_size.split(',')))
+    batch_size = world_size - 2  # No. of observers
+
+    os.environ['MASTER_ADDR'] = master_addr
+    os.environ['MASTER_PORT'] = master_port
+    print("running run worker")
+    if rank == 0:
+        rpc.init_rpc(COORDINATOR_NAME, rank=rank, world_size=world_size)
+
+        coordinator = CoordinatorBase(
+            batch_size, batch, state_size, nlayers, out_features)
+        coordinator.run_coordinator(TOTAL_EPISODES, TOTAL_EPISODE_STEPS, queue)
+
+
+    elif rank == 1:
+        rpc.init_rpc(AGENT_NAME, rank=rank, world_size=world_size)
+    else:
+        rpc.init_rpc(OBSERVER_NAME.format(rank),
+                     rank=rank, world_size=world_size)
+    rpc.shutdown()
+
+
+def main():
+    GRAPH_VARIABLES = {'world_size':[24,48,96,192], 'batch': [True, False], 
+    'state_size': ['10,20,10', '15,20,25'], 'nlayers': [5, 10], 'out_features': [10, 20]}
```

Comment:
rest of the code base would use the following format:

```python
GRAPH_VARIABLES = {
    'world_size' : [24,48,96,192], 
    'batch' :  [True, False], 
    ...
}
```

Reply:
Would I be correct if I assume in the final version, these will be configurable?

Reply:
Originally I had in mind that we would provide a set of variables for each argument deemed useful for users to get an idea of what general outcomes variations in these arguments would produce.  However I agree I think configuration is a good idea.  Since your more familiar with the users, do you think I should add or delete any og the GRAPH_VARIABLES keys?

Reply:
> Since your more familiar with the users, do you think I should add or delete any og the GRAPH_VARIABLES keys?

Since this file will land in PyTorch repo, let's avoid these hard-coded configs. I have no objection of having `graph_variables` as one input arg. In order to avoid these magic numbers, one option can be letting users to pass in a path to a json config file that contains these numbers.

Reply:
Thinking about this again. What `graph_variables` offers is basically allowing users to do `--arg=op1,op2,op3`, and only support this for one arg. If this is your intention, it might be more convenient for users if we directly allow providing multiple args, IIRC, this also aligns with one of your earlier suggestion. I mean sth like this:

```
python launcher.py --world_size=20,40,60,80 --batch=True --nlayers=100
```

Initially, it's OK if we only support at most one of the args to contain multiple options. Later, we can extend that to multiple args and automatically run benchmarks for all combinations of specified arg options.